File: src/razie/base/scripting/TimeOfDay.java
/**  ____    __    ____  ____  ____/___     ____  __  __  ____
 *  (  _ \  /__\  (_   )(_  _)( ___) __)   (  _ \(  )(  )(  _ \           Read
 *   )   / /(__)\  / /_  _)(_  )__)\__ \    )___/ )(__)(  ) _ <     README.txt
 *  (_)\_)(__)(__)(____)(____)(____)___/   (__)  (______)(____/   LICENESE.txt
 */
package razie.base.scripting;

import java.util.Date;
import razie.base.*;

/**
 * easy to use scriptable for date-time related stuff. I use it to play with scripting and presentation etc
 * 
 * @author razvanc
 */
public class TimeOfDay {

    public static String calcvalue() {return new TimeOfDay().value();}
    
    public TimeOfDay() {
        if (aivalues == null) {
            aivalues = new ActionItem[values.length];

            for (int i = 0; i < values.length; i++)
                aivalues[i] = new ActionItem(values[i]);
        }
    }

    public String[] tags() {
        return TAGS;
    }

    public ActionItem[] values() {
        return aivalues;
    }

    public String value() {
        Date dt = new Date();
        return new RangeSel<String>().rangeSel(dt.getHours(), rangei, values, dflt);
    }

    public ActionItem aivalue() {
        Date dt = new Date();
        return new RangeSel<ActionItem>().rangeSel(dt.getHours(), rangei, aivalues, aidflt);
    }

    static String[] TAGS   = { "time" };

    int[][]         rangei = { { 0, 6 }, { 7, 11 }, { 12, 19 }, { 20, 23 } };
    public static String[]        values = { "night", "morning", "day", "evening" };
    String          dflt   = "?";

    ActionItem[]    aivalues;
    ActionItem      aidflt = new ActionItem("?");

    String[][]      ranges = { { "0", "6", "night" }, { "7", "11", "morning" }, { "12", "18", "day" },
            { "19", "23", "evening" } };
}

File: core/src/main/scala/razie/base/scripting/TimeOfDay.scala
/**  ____    __    ____  ____  ____,,___     ____  __  __  ____
 *  (  _ \  /__\  (_   )(_  _)( ___)/ __)   (  _ \(  )(  )(  _ \           Read
 *   )   / /(__)\  / /_  _)(_  )__) \__ \    )___/ )(__)(  ) _ <     README.txt
 *  (_)\_)(__)(__)(____)(____)(____)(___/   (__)  (______)(____/    LICENSE.txt
 */
package razie.base.scripting;

import java.util.Date;
import razie.base._

/**
 * easy to use scriptable for date-time related stuff. I use it to play with scripting and presentation etc
 * 
 * @author razvanc
 */
class TimeOfDay {

  if (aivalues == null)
    aivalues = {
      for (i <- 0 to values.length)
        yield values()(i)
    }.toArray

  def tags() = TimeOfDay.TAGS

  def values() = aivalues

  def value() = {
    val dt = new Date()
    RangeSel.rangeSel(dt.getHours(), TimeOfDay.rangei, TimeOfDay.values, dflt);
  }

  def aivalue() = {
    val dt = new Date()
    RangeSel.rangeSel(dt.getHours(), TimeOfDay.rangei, aivalues, aidflt);
  }

  val dflt = "?";

  var aivalues: Array[ActionItem] = null
  val aidflt = new ActionItem("?");

}

object TimeOfDay {

  //    def calcvalue() : String = {new TimeOfDay().value();}

  val TAGS = Array("time")

  val rangei: Array[Array[Int]] = Array(Array(0, 6), Array(7, 11), Array(12, 19), Array(20, 23))
  val values = Array("night", "morning", "day", "evening")

  val ranges: Array[Array[String]] = Array(Array("0", "6", "night"), Array("7", "11", "morning"), Array("12", "18", "day"),
    Array("19", "23", "evening"))
}
